docs/development/design/privacy-levels.rst

```diff
@@ -0,0 +1,206 @@
+
+#. Public
+#. Private
+#. Protected (currently hidden)
+
+These levels of privacy aren't clear and bring confusion to our users.
+Also, the private level doesn't makes sense on the community site,
+since we only support public projects.
+
+Places where we use the privacy levels are:
+
+- On serving docs
+- Footer
+- Dashboard
+
+Project level privacy
+---------------------
+
+Project level privacy was meant to control the dashboard visibility.
+
+This privacy level brings to confusion when users want to make a version public.
+We should remove all the project privacy levels.
+
+For the community site the dashboard would be always visible,
+and for the commercial site, the dashboard would be always hidden.
+
+The project privacy level is also used to serve the ``404.html`` page,
+show ``robots.txt``, and show  ``sitemap.xml``.
+The privacy level of the default version should be used instead.
+
+Some other ideas about keeping the privacy level is to dictate the default version level of new versions,
+but removing all other logic related to this privacy level.
+This can be (or is going to be) possible with automation rules,
+so we can just remove the field.
+
+Version level privacy
+---------------------
+
+Version level privacy is mainly used to restrict access to documentation.
+For public level, everyone can access to the documentation.
+For private level, only users that are maintainers or that belong to a team with access
+(for the commercial site)
+can access to the documentation.
+
+The protected privacy level was meant to hid versions from listings and search.
+For the community site these versions are treated like public versions,
+and on the commercial site they are treated like private.
+
+The protected privacy level is currently hidden.
+To keep the behavior of hid the versions from listings and search,
+a new field should be added to the Version model: ``hidden``.
+The privacy level (public or private) would be respected to determinate access to the documentation.
+
+For the community site, the privacy level would be public and can't be changed.
+
+The default privacy level of new versions for the commercial site would be ``private``.
+
+Footer
+------
+
+The footer is used to display not hidden versions that the current user has access to.
+
+For the community site not changes are required on the footer.
+
+For the commercial site we use the project level privacy to decide if show or not
+links to the project's dashboard: downloads, project home, and builds.
+Given that the project privacy level would be removed (and the dashboard is always under login),
+those links would never be shown (except for admin users).
+
+Overview
+--------
+
+For the community site:
+
+- The project's dashboard is visible to all users.
+- All versions are always public.
+- The footer shows links to the project's dashboard (build, downloads, home) to all users.
+- Only versions with ``hidden = False`` are listed on the footer and appear on search results.
```

Comment:
How does somebody set `hidden=True` for versions? Would that just be in the usual version form? Does that make sense on .org? Is the idea perhaps for some beta version that is fine for people to access but they don't want it in search results?

Reply:
Yes, it would be just another field on the form. And it makes sense on .org, below are some uses cases that this attribute will replace (protected versions)